.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  - repo: local
+    hooks:
+      - id: format-code
+        name: Format Code
+        entry: ./dev/format.sh
+        language: script
+        # Ensures the script runs from the repository root:
+        pass_filenames: false
+        stages: [commit]
```

Comment:
We should only run `test.sh`. Imagine the following scenario:
- Stashed files are not properly formatted
- Developer runs git commit
- Pre-commit runs `format.sh` -> files are now properly formatted
- Pre-commit runs `test.sh` -> passes, since files are properly formatted
- `git commit` still commits files that have the wrong formatting since the changes caused by `format.sh` are not staged for commit

Reply:
@danieljanes once `format.sh` runs, pre-commit works in a way that if there was something changed by `format.sh`, the commit is blocked from being committed.
<img width="647" alt="Screenshot 2024-03-28 at 13 54 49" src="https://github.com/adap/flower/assets/71217662/54bfa64e-0eaa-4bd4-8654-97e72ddec749">

Reply:
You can then add the new changes caused by `format.sh` and then commit again.

Reply:
Oh nice! Let's keep it as-is then.